packages/medusa/src/api-v2/auth/[scope]/[auth_provider]/callback/route.ts

```diff
@@ -20,18 +20,26 @@ export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
     protocol: req.protocol,
   } as AuthenticationInput
 
-  const authResult = await service.validateCallback(authProvider, authData)
+  const authResult = await service.validateCallback(auth_provider, authData)
 
-  const { success, error, authUser, location } = authResult
-  if (location) {
-    res.redirect(location)
-    return
-  }
+  const { success, error, authUser, successRedirectUrl } = authResult
 
   if (success) {
     const { jwt_secret } = req.scope.resolve("configModule").projectConfig
-    const token = jwt.sign(authUser, jwt_secret)
-    return res.status(200).json({ token })
+
+    const user = { ...authUser }
```

Comment:
Why do we need to destructure the authUser here?

Reply:
The payload to `jwt.sign` needs to be a plain object and it was being returned as an entity object here. I decided to serialize the auth user response at the service level in 8e8681a50be0d70a9dfdcddafd0238c0a1cc1ac3, so we don't need this dirty destructuring.

Reply:
@olivermrbl ah I see, yeah that sound better indeed